openlibrary/catalog/marc/tests/test_data/bin_expect/880_arabic_french_many_linkages.json

```diff
@@ -0,0 +1,72 @@
+{
+  "publish_date": "2009",
+  "publish_country": "mr",
+  "languages": [
+    "ara",
+    "fre"
+  ],
+  "oclc_numbers": [
+    "672263227"
+  ],
+  "other_titles": [
+    "Intiqāl al-afkār wa-al-taqnīyāt fī al-Maghārib wa-al-ʻālam al-mutawassiṭī"
+  ],
+  "edition_name": "al-Ṭabʻah 1.",
+  "series": [
+    "Silsilat nadawāt wa-munāẓarāt -- raqm 160",
+    "Manshūrāt Kullīyat al-Ādāb wa-al-ʻUlūm al-Insānīyah bi-al-Rabāṭ -- raqm 160."
+  ],
+  "notes": "Includes bibliographical references.\n\nArabic and French.",
+  "authors": [
+    {
+      "name": "El Moudden, Abderrahmane",
+      "entity_type": "person",
+      "personal_name": "El Moudden, Abderrahmane",
+      "alternate_names": [
+	"مودن، عبد الرحيم"
```

Comment:
Unfortunately the source MARC has confused the last-names of two of the authors, even though the code is now importing the linkages correctly.  El Moudden, Abderrahmane's last name should be عبد الرحمن  not  عبد الرحيم (Rahman, not Rahim), which is the name of another contributor, in an adjacent 880 field.

```
880 1  $6700-05/(3/r$aمودن، عبد الرحيم.
880 1  $6700-06/(3/r$aبنحادة، عبد الرحيم.
```

The alt script by statement has the authors correctly:
تنسيق عبد الرحمن المودن، عبد الرحيم بنحادة، محمد الأزهر الغربي

I didn't notice the problem with the source data until after I had the name importing working.

Also, why are the other 700 field contributors not being imported as authors in the same way as the 700s from the Nihon no chasho example? 

It looks like in the `100` + `700`s case, only the `100` individual is made an author, the `700`s are contributors.

In the _only_ `700`s case, each `700` is added as an equal author.

Is this desired behavior?

Reply:
It looks like treating `700`s as contributors unless there is no main`1xx` entries is deliberate behavior. I can't find anything that confirms either way that this is a correct or incorrect assumption.  `700`s seem flexible, and although there is provision for (multiple?) subfields to state the exact relationship of the name to the record -- https://www.loc.gov/marc/bibliographic/bd700.html  these don't have to exist, and in practice often don't.
It seems like it works out, but there is a risk some contributors (illustrators / translators) might get added as authors, and conversely some equally  responsible authors may get added as mere contributors.

It looks like there isn't a clear way to indicate all the possibilities in MARC, or at least actual cataloging practice varies considerably.

I won't change the `1xx` / `7xx` behavior in this PR.  If a field is picked as an `author` rather than in the `contributions` list, and an `880` alternate script version exists, it will now be picked up if it is `1xx` or `7xx`.

Reply:
Can you create a separate issue for the 100+700 case? I suspect that the current behavior isn't correct, but can't easily verify and don't have time right now to do extensive research.